Thermald lite

cyxlily opened the pull request.

sysopenci commented:
Improper Commit Message
Tracked on not found in commit message,
    make sure Tracked-On: Jira-ticket  is present.

sysopenci commented:
Improper Commit Message
Tracked on not found in commit message,
    make sure Tracked-On: Jira-ticket  is present.

Commit:
Add sepolicy for thermald-lite

Add sepolicy for thermald-lite.

Tracked-On: OAM-116981
Signed-off-by: Cui, Yuxin <yuxin.cui@intel.com>

sysopenci commented:
Improper Commit Message
Jira issue not in ['bug','improvement'],
    make sure jira issue is of type bug or improvement.

sysopenci commented:
Improper Commit Message
Jira issue not in ['bug','improvement'],
    make sure jira issue is of type bug or improvement.

Commit:
Add sepolicy for thermald-lite log switch

Add sepolicy for thermald-lite log switch
persist.vendor.thermald_lite.logs.enabled

Tracked-On: OAM-116981
Signed-off-by: Cui, Yuxin <yuxin.cui@intel.com>

Commit:
Add init sepolicy for thermald-lite

Add init sepolicy for thermald-lite, add sepolicy for path search,
remove some unused rules, and polish some rules.

Tracked-On: OAM-116981
Signed-off-by: Cui, Yuxin <yuxin.cui@intel.com>

ceiba1985 commented on thermal/thermal-daemon/thermald-lite.te:
```diff
@@ -2,10 +2,14 @@
 type thermald-lite, domain;
 type thermald-lite_exec, system_file_type, exec_type, file_type;
 
+allow init thermald-lite:process transition;
```
this commit is not needed

ceiba1985 commented on thermal/thermal-daemon/thermald-lite.te:
```diff
@@ -0,0 +1,10 @@
+# thermald-lite seclabel is specified in init.<board>.rc
+type thermald-lite, domain;
+type thermald-lite_exec, system_file_type, exec_type, file_type;
```
delete this line

ceiba1985 commented on thermal/thermal-daemon/thermald-lite.te:
```diff
@@ -0,0 +1,10 @@
+allow thermald-lite sysfs_thermal:dir search;
+allow thermald-lite sysfs_thermal:file { getattr open read };
+
+allow thermald-lite kmsg_device:chr_file rw_file_perms;
```
why you need to read/write kmsg_device?

ceiba1985 commented on thermal/thermal-daemon/thermald-lite.te:
```diff
@@ -0,0 +1,10 @@
+allow thermald-lite sysfs_thermal:file { getattr open read };
+
+allow thermald-lite kmsg_device:chr_file rw_file_perms;
+allow thermald-lite rootfs:file entrypoint;
```
delete this line

ceiba1985 commented on thermal/thermal-daemon/thermald-lite.te:
```diff
@@ -0,0 +1,10 @@
+# thermald-lite seclabel is specified in init.<board>.rc
+type thermald-lite, domain;
```
please refer to this te file and add "domain_trans(init, rootfs, thermal_lite);". And you'd better wrap the rules with recovery_only.
http://inlubt0291.iind.intel.com:8080/source/xref/Android-S-GP2.3/device/intel/sepolicy/thermal/dptf/thermal_lite.te?r=71253d20#7

ceiba1985 commented on thermal/thermal-daemon/thermald-lite.te:
```diff
@@ -0,0 +1,20 @@
+recovery_only(`
+domain_trans(init, rootfs, thermald-lite);
+
+allow init thermald-lite:process { rlimitinh siginh };
```
remove this line

ceiba1985 commented on thermal/thermal-daemon/thermald-lite.te:
```diff
@@ -0,0 +1,20 @@
+domain_trans(init, rootfs, thermald-lite);
+
+allow init thermald-lite:process { rlimitinh siginh };
+allow thermald-lite sysfs_thermal:dir { search open read };
```
replace { search open read } with r_dir_perms

ceiba1985 commented on thermal/thermal-daemon/thermald-lite.te:
```diff
@@ -0,0 +1,20 @@
+
+allow init thermald-lite:process { rlimitinh siginh };
+allow thermald-lite sysfs_thermal:dir { search open read };
+allow thermald-lite sysfs_thermal:file { getattr open read };
```
replace { getattr open read } with r_file_perms

ceiba1985 commented on thermal/thermal-daemon/thermald-lite.te:
```diff
@@ -0,0 +1,20 @@
+allow thermald-lite sysfs_thermal:file { getattr open read };
+allow thermald-lite rootfs:dir read;
+allow thermald-lite rootfs:file { map read };
+allow thermald-lite sysfs_thermal_management:dir { search open read };
```
r_dir_perms

ceiba1985 commented on thermal/thermal-daemon/thermald-lite.te:
```diff
@@ -0,0 +1,20 @@
+allow thermald-lite rootfs:dir read;
+allow thermald-lite rootfs:file { map read };
+allow thermald-lite sysfs_thermal_management:dir { search open read };
+allow thermald-lite sysfs_thermal_management:file { getattr open read };
```
r_file_perms

ceiba1985 commented on thermal/thermal-daemon/thermald-lite.te:
```diff
@@ -0,0 +1,20 @@
+allow thermald-lite vendor_thermal_prop:property_service set;
+
+# properties
+get_prop(thermald-lite, vendor_thermald_lite_logs_enabled_prop)
```
remove this line

sysopenci commented:
Android CI has started Engineering Build for this issue ,Please check the linked Tracked-On issue/Android CI Web  for more details.

sysopenci commented:
SUCCESS: Android CI has completed Engineering Build for this issue.Please check the linked Tracked-On issue/Android CI Web  for more details.

sysopenci commented:
Android CI has started Engineering Build for this issue ,Please check the linked Tracked-On issue/Android CI Web  for more details.

sysopenci commented:
SUCCESS: Android CI has completed Engineering Build for this issue.Please check the linked Tracked-On issue/Android CI Web  for more details.

sysopenci commented:
Android CI has started MERGE Build for this pr ,Please check the linked Tracked-On issue/Android CI Web  for more details.

sysopenci commented:
Android CI has completed MERGE Build for this pr, build is SUCCESS. Please check the linked Tracked-On issue/Android CI Web  for more details. For Binaries: /cactus-absp-or-local/celadon_smr0_apollo-merge/103
